src/components/modal.js

```diff
@@ -0,0 +1,18 @@
+import A11yDialog from "a11y-dialog";
+
+/**
+ *@param {string } modalClass
+ * @param {string} contentId
+ * @param {function} onShow
+ */
+
+const init = (modalClass, onShow = () => {}) => {
+  const modalElms = document.querySelectorAll(`[class*=${modalClass}]`);
```

Comment:
I just thought about this last night, i think you need to wrap all the contents of the init function in this https://developer.mozilla.org/en-US/docs/Web/API/Window/DOMContentLoaded_event.

This will make sure all the elements are loaded on the page, before we execute the javascript. If we don't do it, its possible the script will execute before all the modal html is loaded.